openedx/tests/xblock_integration/test_recommender.py

```diff
@@ -650,7 +651,10 @@ def attempt_upload_file_and_verify_result(self, test_case, event_name, content=N
         happens or is rejected as expected.
         """
         if 'magic_number' in test_case:
-            f_handler = StringIO(test_case['magic_number'].decode('hex'))
+            if six.PY2:
+                f_handler = StringIO(codecs.decode(test_case['magic_number'], 'hex_codec'))
+            else:
+                f_handler = StringIO(codecs.decode(test_case['magic_number'], 'hex_codec').decode('utf-8'))
```

Comment:
`decode` should work in both python 2 and 3 here shouldn't it?

Reply:
But isn't the decode here going to just cause f_handler to be a string instead of a stream?  I think you may want https://six.readthedocs.io/#six.BytesIO instead.

Reply:
@feanil yes we are already using this for python3. But the StringIO for python 2 is different that is why I had to add this condition